x-pack/legacy/plugins/maps/public/layers/styles/vector/components/legend/vector_style_legend.js

```diff
@@ -30,27 +27,26 @@ export class VectorStyleLegend extends Component {
   }
 
   _loadRows = _.debounce(async () => {
-    const rows = await this.props.loadRows();
-    const rowDescriptors = rows.map(row => {
+    const styles = await this.props.stylesPromise;
+    const styleDescriptorPromises = styles.map(async style => {
       return {
-        label: row.label,
-        range: row.meta,
-        styleOptions: row.style.getOptions(),
+        type: style.getStyleName(),
+        options: style.getOptions(),
+        fieldMeta: style.getFieldMeta(),
+        label: await style.getField().getLabel(),
```

Comment:
Do not need to include `label` in descriptor diff since `DynamicLegendRow` manages label in its internal state.

Reply:
discussed in person. keeping as is now.

this is required to have the label update with type-as-you-go functionality. The async-nature of the legend-rendering can be removed, once determination of the geometry-type determination is made sync.